sdks/java/core/src/main/java/org/apache/beam/sdk/io/CountingSource.java

```diff
@@ -83,6 +83,20 @@
   }
 
   /**
+   * Creates a {@link BoundedSource} that will produce the specified number of elements,
+   * from {@code startOffset} to {@code numElements - 1}.
+   *
+   * @deprecated use {@link CountingInput#forSubrange(long, long)} instead
+   */
+  @Deprecated
+  static BoundedSource<Long> createSourceForSubrange(long startOffset, long endOffset) {
+    checkArgument(endOffset > startOffset, "numElements (%s) must be greater than 0",
```

Comment:
It's better to mention in an error message the things that the user directly specifies. Referring to numElements implies that the user specified a parameter called numElements, which is not the case. Rephrase as "endOffset must be greater than startOffset"?